File: url-shortener-backend/src/url/entity/url.entity.ts
import { User } from 'src/auth/entity/user.entity';
import { Column, CreateDateColumn, Entity, ManyToOne, PrimaryGeneratedColumn, Unique, UpdateDateColumn } from 'typeorm';

@Entity()
export class Url {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  url: string;

  @Column({unique: true})
  short: string;

  @Column()
  clicks: number

  @ManyToOne(() => User, (user) => user.urls, {nullable: true})
  user: User

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}

File: url-shortener-backend-go/model/url_entity.go
package model

import "time"

type Url struct {
	// gorm.Model
	ID        uint `gorm:"primarykey"`
	CreatedAt time.Time
	UpdatedAt time.Time
	url       string `gorm:"size:255;not null"`
	short     string `gorm:"size:255;not null"`
	clicks    uint   `gorm:"-;default:0"`
	User      User
	UserId    uint
}
